### PR TITLE
Remove Newtonsoft.Json.Schema dependency

### PR DESCRIPTION
**Pull request by eerhardt**
We are already using JsonSchema.Net in the components tests. We don't need to have both ways of doing JSON Schema in the repo.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4709)

### CODE REVIEW

File: tests/Aspire.Hosting.Tests/Schema/SchemaTests.cs [hunk 1]
```diff
@@ -249,13 +242,13 @@ public async Task ManifestAcceptsUnknownResourceType()
             }
             """;
 
-        var manifestJson = JToken.Parse(manifestTest);
-        var schema = await GetSchemaAsync();
-        Assert.True(manifestJson.IsValid(schema));
+        var manifestJson = JsonNode.Parse(manifestTest);
+        var schema = GetSchema();
+        Assert.True(schema.Evaluate(manifestJson).IsValid);
```
Comment (radical) [inline tests/Aspire.Hosting.Tests/Schema/SchemaTests.cs:247]: Just curious - why not use `AssertValid(..)` everywhere?

File: tests/Aspire.Hosting.Tests/Schema/SchemaTests.cs [hunk 2]
```diff
@@ -212,32 +211,26 @@ public async Task ValidateApplicationSamples(string testCaseName, Action<IDistri
...
+        var manifestJson = JsonNode.Parse(manifestText);
+        var schema = GetSchema();
+        AssertValid(manifestJson, schema);
     }
 
     [Fact]
-    public async Task SchemaRejectsEmptyManifest()
+    public void SchemaRejectsEmptyManifest()
     {
         var manifestTest = """
```
Comment (radical) [inline tests/Aspire.Hosting.Tests/Schema/SchemaTests.cs:222]: super duper unrelated nit: `manifestTest` -> `manifestText`?

**Review by radical — APPROVED** [review]
Left some comments. Other than that LGTM!

File: tests/Aspire.Hosting.Tests/Schema/SchemaTests.cs (hunk 1 above)
Comment (eerhardt) [reply tests/Aspire.Hosting.Tests/Schema/SchemaTests.cs:247]: I guess I was just migrating the code as it is. I can change it though.

**Pull request merged**